tests/casper/langs/test_state_lang.py

```diff
@@ -0,0 +1,292 @@
+"""The language testing module ... """
+import pytest
+
+from simulations.state_language import StateLanguage
+
+from casper.protocols.blockchain.blockchain_protocol import BlockchainProtocol
+from casper.protocols.binary.binary_protocol import BinaryProtocol
+from casper.protocols.order.order_protocol import OrderProtocol
+from casper.protocols.integer.integer_protocol import IntegerProtocol
+
+from casper.network import Network
+
+PROTOCOLS = [BlockchainProtocol, BinaryProtocol, OrderProtocol, IntegerProtocol]
+# protocols with a single genesis block
+GENESIS_PROTOCOLS = [BlockchainProtocol]
+# each val starts with different block
+INITAL_MESSAGE_PROTOCOLS = [BinaryProtocol, OrderProtocol, IntegerProtocol]
+
+
+@pytest.mark.parametrize(
```

Comment:
define `protocol_params` at the top of the file:

`protocol_params = [(protocol) for protocol in PROTOCOLS]`

 and then just use this line each time you want to pass in just protocols to a test:

`@pytest.mark.parametrize('protocol', protocol_params)`